catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/builder/JsonModelBuilder.java

```diff
@@ -102,129 +82,271 @@ public FilterNode getResult() {
-    verifyResultNotYetRetrieved();
-    verifyTerminalNodeNotInProgress();
-    verifyLogicalNodeInProgress();
-    verifyLogicalNodeHasChildren();
-    depth.pop();
+  public JsonModelBuilder or() {
+    beginBinaryLogicType("OR");
     return this;
   }
 
-  // Note: Currently taking in the XML local part as the "operator"
   @Override
-  public JsonModelBuilder beginBinaryComparisonType(String operator) {
-    verifyResultNotYetRetrieved();
-    verifyTerminalNodeNotInProgress();
-    String jsonOperator = BINARY_COMPARE_MAPPING.get(operator);
-    if (jsonOperator == null) {
-      throw new IllegalArgumentException(
-          "Cannot find mapping for binary comparison operator: " + operator);
+  public JsonModelBuilder end() {
+    if (supplierInProgress != null) {
+      if (supplierInProgress.getParent() != null) {
+        endFunctionType();
+        return this;
+      }
+      endTerminalType();
+      return this;
     }
-    nodeInProgress = new FilterNodeImpl(jsonOperator);
+    endBinaryLogicType();
     return this;
   }
 
   @Override
-  public FlatFilterBuilder beginPropertyIsLikeType(String operator, boolean matchCase) {
-    verifyResultNotYetRetrieved();
-    verifyTerminalNodeNotInProgress();
-    if (!PROPERTY_IS_LIKE.equals(operator)) {
-      throw new IllegalArgumentException("Cannot find mapping for like operator: " + operator);
-    }
-    // For now, will always choose ILIKE
-    String jsonOperator = (matchCase) ? LIKE : ILIKE;
-    nodeInProgress = new FilterNodeImpl(jsonOperator);
+  public JsonModelBuilder isEqualTo(boolean matchCase) {
+    beginTerminalType("=");
     return this;
   }
 
   @Override
-  public FlatFilterBuilder beginBinaryTemporalType(String operator) {
-    verifyResultNotYetRetrieved();
-    verifyTerminalNodeNotInProgress();
-    String jsonOperator = BINARY_TEMPORAL_MAPPING.get(operator);
-    if (jsonOperator == null) {
-      throw new IllegalArgumentException(
-          "Cannot find mapping for binary temporal operator: " + operator);
-    }
-    nodeInProgress = new FilterNodeImpl(jsonOperator);
+  public JsonModelBuilder isNotEqualTo(boolean matchCase) {
+    beginTerminalType("!=");
     return this;
   }
 
   @Override
-  public JsonModelBuilder beginBinarySpatialType(String operator) {
+  public JsonModelBuilder isGreaterThan(boolean matchCase) {
+    beginTerminalType(">");
+    return this;
+  }
+
+  @Override
+  public JsonModelBuilder isGreaterThanOrEqualTo(boolean matchCase) {
+    beginTerminalType(">=");
+    return this;
+  }
+
+  @Override
+  public JsonModelBuilder isLessThan(boolean matchCase) {
+    beginTerminalType("<");
+    return this;
+  }
+
+  @Override
+  public JsonModelBuilder isLessThanOrEqualTo(boolean matchCase) {
+    beginTerminalType("<=");
+    return this;
+  }
+
+  @Override
+  public JsonModelBuilder like(
+      boolean matchCase, String wildcard, String singleChar, String escape) {
+    String operator = (matchCase) ? "LIKE" : "ILIKE";
```

Comment:
✏️ Be sure to test matchCase options on search forms. This alone might not be enough.

Reply:
Clarification: MATCHCASE & NEAR still unsupported. 
Ticket for adding support - https://codice.atlassian.net/browse/DDF-3829